File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        vals = {}
        for i, x in enumerate(nums):
            if vals.get(target-x) != None:
                res = []
                res.append(vals.get(target-x))
                res.append(i)
                return res
            else:
                vals[x] = i

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int tar = target - nums[i];
            if (mp.containsKey(tar)) {
                int[] ans = { mp.get(tar), i };
                return ans;
            } else {
                mp.put(nums[i], i);
            }
        }
        int[] ans = new int[2];
        return ans;
    }
}